.circleci/config.yml

```diff
@@ -45,8 +45,8 @@ jobs:
               - "node_modules"
 
       - run:
-          name: Npm lint
-          command:  npm run lint
+          name: Prettier
+          command:  npm run format:test
```

Comment:
FYI @valentijnnieman I added this formatting commit. I noticed some linting issues and was surprised the tests didn't catch them... turns out
- `npm run lint` as currently defined is a fixer, not a tester, so it can't fail. Anyway we want to test `prettier` since it's stricter (is there actually a purpose to `eslint` if we're running `prettier` too?)
- `format` and `format:test` didn't inspect `.react.js` files - I guess `*.js` doesn't match `*.react.js`. I fixed that in `package.json`.

Reply:
Oh I see... `prettier` only does formatting, not code quality checks, didn't realize that. I'll update so this runs both (and both as tests!)

Reply:
`prettier` can test code formatting too, if that's what we want. I set that up in core components I believe

Reply:
As here: https://github.com/plotly/dash-core-components/blob/83a3ea07fd8af9ee3ca57c8da8ed32483e96f87c/package.json#L25

Reply:
Hmm, they seem to say otherwise https://prettier.io/docs/en/comparison.html

> Code-quality rules: eg no-unused-vars, no-extra-bind, no-implicit-globals, prefer-promise-reject-errors...
> 
> Prettier does nothing to help with those kind of rules. They are also the most important ones provided by linters as they are likely to catch real bugs with your code!

Anyway I set it to run both https://github.com/plotly/dash-renderer/pull/75/commits/6c309d714de747b324e109b5ccaa76ad2e9fdbb4

BTW it would be worth looking at this `.react.js` issue in the other repos...

Also in the main `dash` repo I also made all of these tests run to completion, but I didn't do that here, this setup will still quit on first failure. I prefer to get all my errors at once, but seems like I may be in the minority on that issue...

Reply:
Yeah, I was talking about testing formatting - it'll throw an error if the code isn't formatted the way `prettier` wants it. I like it personally - that way I can actually use `prettier` and not worry about committing lots of noise regarding formatting changes. But I know others aren't so keen on it! Anyway, yeah, it doesn't replace linting or anything. 

Not sure what you mean about the tests - aren't these ones all being run to completion too?

Reply:
The way I did it in https://github.com/plotly/dash-renderer/pull/75/commits/6c309d714de747b324e109b5ccaa76ad2e9fdbb4, CI runs `lint:test`, and if that fails the whole test run stops. If it succeeds we go on to `format:test`, and if that fails the whole test run stops. Only if both of those pass do we even run the python tests.

`dash`, on the other hand, is currently [set up](https://github.com/plotly/dash/blob/master/test.sh) to run all of these no matter what, but still registering a failure if any of them failed. I prefer that, so I can go back and fix all the errors at once, but seems like others mostly prefer to get an answer more quickly.